File: quiz-frontend/src/components/QuizResult.js
import React from "react";
import quizApi from "../api/quizApi";

const res = await quizApi.get("/quiz/start?count=3");


function QuizResult({ result }) {
  return (
    <div>
      <h2>Result</h2>
      <p>Total Questions: {result.totalQuestions}</p>
      <p>Correct Answers: {result.correctAnswers}</p>
      <p>Score: {result.score}%</p>
    </div>
  );
}

export default QuizResult;

File: src/main/java/com/example/quiz/model/QuizResult.java
package com.example.quiz.model;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "quiz_results")
public class QuizResult {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long userId;          // ✅ who took the quiz
    private int totalQuestions;   // ✅ total questions attempted
    private int correctAnswers;   // ✅ number of correct answers
    private int score;            // ✅ percentage score
}
